### PR TITLE
network.tf file to create networking resources for Splunk export

### PR DESCRIPTION
**Pull request by npredey**
* Also ran `terraform fmt` to clean up files

### CODE REVIEW

**Review by rarsan — CHANGES_REQUESTED** [review]
Thank you for this contribution!
Looks great, a couple of comments below. 
Possible to remove whitespace changes to preserve history?

File: network.tf [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+resource "google_compute_router_nat" "dataflow_nat" {
+  count = var.create_network == true ? 1 : 0
+
+  name                               = "${var.network}-${var.region}-router-nat"
+  router                             = google_compute_router.dataflow_to_splunk_router.name
+  region                             = google_compute_router.dataflow_to_splunk_router.region
+  nat_ip_allocate_option             = "MANUAL_ONLY"
+  source_subnetwork_ip_ranges_to_nat = "LIST_OF_SUBNETWORKS"
+  nat_ips                            = google_compute_address.dataflow_nat_ip_address.*.self_link
+  min_ports_per_vm                   = 1024
```
[the remark this answers is not in the document]
Comment (npredey) [reply network.tf:50]: This is a good call out. I will change this to 128.

File: network.tf [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+  name                    = var.network
+  auto_create_subnetworks = false
+}
+
+resource "google_compute_subnetwork" "splunk_subnet" {
+  count = var.create_network == true ? 1 : 0
+
+  name                     = local.subnet_name
+  ip_cidr_range            = var.primary_subnet_cidr
```
[the remark this answers is not in the document]
Comment (npredey) [reply network.tf:12]: I'll add it in and then create that range as the default.

**Comment by npredey** [comment]
> Thank you for this contribution!
> Looks great, a couple of comments below.
> Possible to remove whitespace changes to preserve history?

I was so in the mode of running `terraform fmt` that I committed it without realizing. I can look to revert that.

File: pipeline.tf [hunk 3]
```diff
@@ -49,16 +49,16 @@ resource "google_dataflow_job" "dataflow_job" {
...
+    url       = var.splunk_hec_url
+    token     = var.splunk_hec_token
     parallelism = var.dataflow_job_parallelism
     batchCount = var.dataflow_job_batch_count
     includePubsubMessage = local.dataflow_job_include_pubsub_message
     disableCertificateValidation = var.dataflow_job_disable_certificate_validation
   }
   region = var.region
-  network = var.network
+  network = var.create_netowork == true ? google_compute_network.splunk_export.id : var.network
```
Comment (rarsan) [inline pipeline.tf:62]: just noticed typo with var name.

**Review by rarsan — CHANGES_REQUESTED** [review]
Looks good. Mind adding changes in the README? Particularly new user-facing variables.
Sorry forgot to mention that last time.
Also there's a simple variables placeholder file `variables.yaml` that needs to be updated with these new vars. It's made available for end-users to seed a new user-specific `terraform.tfvars` as described in README setup section.

**Review by rarsan — APPROVED** [review]

**Pull request merged**